systems/console-bff/subscriptions/resolvers/resolver.ts

```diff
@@ -335,271 +335,329 @@ class SubscriptionsResolvers {
-              avg = sum / res.values.length;
-            } else {
-              const sum = res.values.reduce((acc, val) => acc + val[1], 0);
-              avg = sum / res.values.length;
-            }
-          }
-
-          metrics.metrics.push({
-            msg: res.msg,
-            type: res.type,
-            siteId: res.siteId || siteId || "",
-            nodeId: "",
-            success: res.success,
-            value: formatKPIValue(res.type, avg),
-          });
-        }
-      } catch (error) {
-        logger.error(
-          `Error processing site metrics for site ${siteId}: ${error}`
-        );
-        for (const key of metricKeys) {
-          metrics.metrics.push({
+  private async fetchSiteMetrics(
+    baseURL: string,
+    metricKeys: string[],
+    data: GetMetricsSiteStatInput,
+    siteId: string
+  ): Promise<any[]> {
+    const results = await Promise.all(
+      metricKeys.map(async key => {
+        try {
+          return await getSiteMetricRange(baseURL, key, { ...data, siteId });
+        } catch (error) {
+          logger.error(
+            `Error processing site metric ${key} for site ${siteId}: ${error}`
+          );
+          return {
             msg: "Error",
             type: key,
             siteId: siteId || "",
             nodeId: "",
             success: false,
-            value: 0,
-          });
+            values: [],
+          };
         }
-      }
+      })
+    );
+    return results;
+  }
 
-      if (Array.isArray(nodeIds) && nodeIds.length > 0) {
-        for (const nodeId of nodeIds) {
-          try {
-            const metricsKey = getGraphsKeyByType(type);
-            const nodeResults: MetricsStateRes = { metrics: [] };
-
-            if (metricsKey.length > 0) {
-              const metricPromises = metricsKey.map(async key => {
-                const res = await getNodeMetricRange(baseURL, key, { ...data });
-                if (Array.isArray(res.metrics)) {
-                  res.metrics.forEach(metric => {
-                    metrics.metrics.push(processMetricResult(metric));
-                  });
-                }
-              });
+  private processSiteMetricResults(results: any[], siteId: string): any[] {
+    return results.map(res => {
+      let avg = null;
+      let success = res.success;
+
+      if (Array.isArray(res.values) && res.values.length > 0) {
+        if (res.values.length === 1 || res.type === "site_uptime_seconds") {
+          avg = res.values[res.values.length - 1][1];
+        } else if (res.type === "site_uptime_percentage") {
+          const sum = res.values.reduce(
+            (acc: number, val: any) => acc + val[1],
+            0
+          );
+          avg = sum / res.values.length;
+        } else {
+          const sum = res.values.reduce(
+            (acc: number, val: any) => acc + val[1],
+            0
+          );
+          avg = sum / res.values.length;
```

Comment:
We don't need to compute this avg now, If you check the input type `GetMetricsStatInput` which takes `operation` input `avg/sum` with deffault value `avg` so you'll always get the avg value.
For ref check: https://github.com/ukama/ukama/blob/be9dde24907bd49be6b840eed83b237cc1cea2fc/systems/console-bff/subscriptions/resolvers/resolver.ts#L204-L222